packages/static-build/src/utils/gatsby.ts

```diff
@@ -2,39 +2,55 @@ import { PackageJson } from '@vercel/build-utils';
 import { promises as fs } from 'fs';
 import * as path from 'path';
 import semver from 'semver';
+import { URL } from 'url';
 import {
   fileExists,
   readPackageJson,
   DeepWriteable,
   writePackageJson,
 } from './_shared';
 
-const PLUGINS = {
-  GATSBY_PLUGIN_VERCEL_ANALYTICS: '@vercel/gatsby-plugin-vercel-analytics',
-  GATSBY_PLUGIN_VERCEL_BUILDER: '@vercel/gatsby-plugin-vercel-builder',
-};
+const PLUGINS = [
+  '@vercel/gatsby-plugin-vercel-analytics',
+  '@vercel/gatsby-plugin-vercel-builder',
+] as const;
+type PluginName = typeof PLUGINS[number];
+
+const PLUGIN_VERSIONS = new Map<PluginName, string>([
+  ['@vercel/gatsby-plugin-vercel-analytics', 'latest'],
+  ['@vercel/gatsby-plugin-vercel-builder', 'latest'],
+]);
```

Comment:
Small nit pick: I think you can remove this Map and the initialization here instead lazily create the URL on line 112

Reply:
SGTM 718930ca48e1a1d605770a23d8ad86496e355df8